ecobee/beam_batch.py

```diff
@@ -0,0 +1,120 @@
+        else:
+            cur_sequence_actions.append(item)
+            cur_seq_len += delta_seconds
+        prev_item = item
+
+    if len(all_sequences) == 0:
+        all_sequences.append(Sequence(cur_seq_len, cur_sequence_actions))
+    all_sequences_sorted = sorted(all_sequences, key=lambda x: x[0], reverse=True)
+    return [(all_sequences_sorted[i].seq_len, [(d.time, d.action) for d in all_sequences_sorted[i].datapoints]) for i in range(0, min(top_n_elems, len(all_sequences_sorted)))]
+
+
+def remove_if_exists(file_names):
+    for file_name in file_names:
+        if os.path.exists(file_name):
+            os.remove(file_name)
+
+
+def run(argv=None, save_main_session=True):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input',
+                        dest='input',
+                        default='test_actions.txt',
+                        help='Input file to process.')
+    parser.add_argument('--output',
+                        dest='output',
+                        default='output.txt',
+                        help='Output file to write results to.')
+    parser.add_argument('--top-n',
+                        default=10,
+                        type = int,
+                        help='Number of top entries to report.')
+    parser.add_argument('--time-period-secs',
+                        default=0,
+                        type=int,
+                        help='If two events are apart by this value of time, they are considered in separate sequences.')
+    parser.add_argument('--interim-results',
+                        default='intermediate.txt',
+                        help='Intermediate file location to write the calculations.')
+    known_args, pipeline_args = parser.parse_known_args(argv)
+    start = time()
+    pipeline_options = PipelineOptions(pipeline_args)
+    pipeline_options.view_as(SetupOptions).save_main_session = save_main_session
+    remove_if_exists([known_args.interim_results, known_args.output])
+
+    p = beam.Pipeline(options=pipeline_options)
+    p | 'read' >> ReadFromText(known_args.input) | 'format' >> beam.Map(to_data_point) \
+    | 'intermediate_write' >> WriteToText(known_args.interim_results,
+                                          num_shards=1, shard_name_template='')
+    result = p.run()
+    result.wait_until_finish()
+
+    batch_events = []
+    with open(known_args.interim_results, 'r') as f:
+        for item in f:
+            batch_events.append(eval(item))
+    sorted_data_points = sorted(batch_events, key=lambda x:datetime.strptime(x.time, date_fmt))
+    output = process(sorted_data_points, known_args.time_period_secs, known_args.top_n)
+    output | 'write' >> WriteToText(known_args.output, num_shards=1, shard_name_template='')
+    os.remove(known_args.interim_results)
+    end = time()
+    print("Total time taken %s seconds" % str(end-start))
+
+
+if __name__ == '__main__':
+    logging.getLogger().setLevel(logging.INFO)
```

Comment:
we are not using loggin anywhere. Should logout some helpful messages about how the job is doing.

Reply:
so apparently, beam uses this configuration internally and produces output in the console. Removing this line is also affecting that. Leaving it here.